include/drivers/flash.h

```diff
@@ -84,9 +84,8 @@ __subsystem struct flash_driver_api {
 /**
  *  @brief  Read data from flash
  *
- *  Most of flash drivers support unaligned flash access, but some have
- *  restrictions on the read offset or/and the read size. Please refer to
- *  the driver implementation to get details on the read alignment requirement.
+ *  All flash drivers support unaligned flash access, without any
+ *  restrictions on the read offset and the read size.
```

Comment:
And what with the nrf_qspi_nor driver? It still [requires](https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/flash/nrf_qspi_nor.c#L487) the size to be a multiple of 4. Moreover, according to the [nRF QSPI documentation](https://infocenter.nordicsemi.com/topic/ps_nrf52840/qspi.html?cp=4_0_0_5_18_2#register.READ.SRC), the read offset needs to be word-aligned, but the driver does not check it nor seems to be handling it in any other way.

Reply:
^^ can this driver be aligned to above expectation?

Reply:
It has be aligned if this PR will be merged.

Reply:
@kamlaz can you post such a patch?